File: code/python/findLucky.py
class Solution(object):
    def findLucky(self, arr):
        """
        :type arr: List[int]
        :rtype: int
        """
        if len(arr) == 0:
            return -1
        dic = {}
        for i in range(len(arr)):
            if arr[i] in dic:
                dic[arr[i]] += 1
            else:
                dic[arr[i]] = 1
        res = -1
        for key in dic:
            if dic[key] == key:
                res = max(key,res)
        return res

File: code/java/findLucky.java
class Solution {
    public int findLucky(int[] arr) {
        if(arr.length == 0)
            return -1;
        Map<Integer,Integer> map = new HashMap<Integer,Integer>();
        for(int i=0;i<arr.length;i++){
            if(map.containsKey(arr[i])){
                int temp = map.get(arr[i]);
                map.replace(arr[i],temp,temp+1);
            }else{
                map.put(arr[i],1);
            }
        }
        int res = -1;
        for(int key: map.keySet()){
            if(map.get(key) == key)
                res = Math.max(res,key);
        }
        return res;
    }
}
